service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java

```diff
@@ -0,0 +1,374 @@
+package cz.muni.fi.pa165.mushrooms.service;
+
+import cz.muni.fi.pa165.mushrooms.dao.MushroomHunterDao;
+import cz.muni.fi.pa165.mushrooms.entity.MushroomHunter;
+import cz.muni.fi.pa165.mushrooms.validation.PersistenceSampleApplicationContext;
+import mockit.Delegate;
+import mockit.Expectations;
+import mockit.Injectable;
+import mockit.Tested;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Created by Matúš on 26.11.2017.
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class MushroomHunterServiceTest extends AbstractTransactionalJUnit4SpringContextTests  {
+
+    @Injectable
+    private MushroomHunterDao mushroomHunterDao;
+
+    @Tested(fullyInitialized = true)
+    private MushroomHunterServiceImpl service;
+
+    private MushroomHunter nonPersistedHunter;
+    private MushroomHunter persistedHunter;
+    private MushroomHunter persistedHunter2;
+    private MushroomHunter newHunter;
+
+    private Map<Long, MushroomHunter> persistedMushroomHunter = new HashMap<>();
+    // stores 'database' size
+    private long dbCounter;
```

Comment:
should start from 1, otherwise you'll get a "persisted" entity with ID 0

Reply:
resolved